oss-fuzz/cbor_load_fuzzer.cc

```diff
@@ -1,20 +1,43 @@
 #include <cstdint>
 #include <cstdio>
+#include <unordered_map>
 
 #include "cbor.h"
 
+static size_t allocated_mem = 0;
+static std::unordered_map<void*, size_t> allocated_len_map;
+
 void *limited_malloc(size_t size) {
-    if (size > 1 << 24) {
+    if (size + allocated_mem > 1 << 24) {
         return nullptr;
     }
-    return malloc(size);
+    void* m = malloc(size);
+    if (m != nullptr) {
+      allocated_mem += size;
+      allocated_len_map[m] = size;
+    }
+    return m;
+}
+
+void limited_free(void *ptr) {
+  if (allocated_len_map.find(ptr) == allocated_len_map.end()) {
```

Comment:
Doesn't this mean there has been a double free? Maybe we should crash here or something (fuzzing has found UB).

Reply:
Makes sense. Do you want me to do another PR or do it yourself?